include/openthread/platform/toolchain.h

```diff
@@ -190,6 +190,45 @@ extern "C" {
 #endif
 
 /**
+ * @def OT_UNUSED_VARIABLE
+ *
+ * Suppress unused variable warning in specific toolchains.
+ *
+ */
+
+/**
+ * @def OT_UNREACHABLE_CODE
+ *
+ * Suppress Unreachable code warning in specific toolchains.
+ *
+ */
+
+#if defined(__ICCARM__)
+
+#define OT_UNUSED_VARIABLE(VARIABLE)    \
+    do                                  \
+    {                                   \
+        if (VARIABLE) {}                \
```

Comment:
Well, I didn't find a pragma that would explicitly tell the compiler that not using a variable is ok, and suppressing a warning, the same as with OT_UNREACHABLE_CODE doesn't work in this case as desired. 

Your proposal looks good to me, though I don't have a chance to check it now, I'll do it on Monday and give a feedback if IAR accept this.